src/tests/suite/daos_aggregate_ec.c

```diff
@@ -0,0 +1,306 @@
+/**
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is for simple tests of EC aggregation. These tests verify the
+ * consistency of the EC data following the completion of the aggregation
+ * step for the data written in each test.
+ *
+ * tests/suite/daos_aggegate_ec.c
+ *
+ *
+ */
+#define D_LOGFAC	DD_FAC(tests)
+
+#include "daos_test.h"
+#include <daos/pool.h>
+#include <daos/mgmt.h>
+#include <daos/container.h>
+#include <daos/event.h>
+
+#define PARITY_INDICATOR (1ULL << 63)
+
+enum {
+	EC_FULL_STRIPE,
+	EC_FULL_CELL,
+	EC_SPECIFIED,
+};
+
+#define assert_success(r) do {\
+	int __rc = (r); \
+	if (__rc != 0) \
+		fail_msg("Not successful!! Error code: " DF_RC, DP_RC(__rc)); \
+	} while (0)
+
+
+/** easily setup an iov and allocate */
+static void
+iov_alloc(d_iov_t *iov, size_t len)
+{
+	D_ALLOC(iov->iov_buf, len);
+	iov->iov_buf_len = iov->iov_len = len;
+}
+
+static void
+iov_alloc_str(d_iov_t *iov, const char *str)
+{
+	iov_alloc(iov, strlen(str) + 1);
+	strcpy(iov->iov_buf, str);
+}
+
+/** daos EC aggregation test context */
+struct ec_agg_test_ctx {
+	/** Pool */
+	daos_handle_t		poh;
+	/** Container */
+	daos_handle_t		coh;
+	daos_cont_info_t	info;
+	uuid_t			uuid;
+	/** Object */
+	daos_handle_t		oh;
+	daos_obj_id_t		oid;
+	daos_key_t		dkey;
+	daos_iod_t		update_iod;
+	d_sg_list_t		update_sgl;
+	daos_iom_t		fetch_iom;
+	daos_iod_t		fetch_iod;
+	d_sg_list_t             fetch_sgl;
+	daos_recx_t             recx[8];
+	daos_recx_t             iom_recx;
+};
+
+inline static char
```

Comment:
(style)  storage class should be at the beginning of the declaration
(style)  inline keyword should sit between storage class and type